Adding GuLoader template

gio-pecora opened the pull request:
Adding GuLoader template with indicators up to 2021-Dec-02

## Description of the change
New Template

## Type of change
New feature (non-breaking change that adds functionality)

Commit:
Adding GuLoader template

Up to 2021-Dec-02

maximelb commented on Emerging_Threats/guloader.yml:
```diff
@@ -0,0 +1,155 @@
+'GuLoader #006':
+  namespace: general
+  detect:
+    event: ONGOING_IDENTITY
```
Those should be targeting `CODE_IDENTITY`. The `ONGOING_IDENTITY` is not reported to the cloud by default.
```suggestion
    event: CODE_IDENTITY
```

gio-pecora replied:
I agree,
I was using the as reference the previous template pushed/reviewed on the repo:
https://github.com/refractionPOINT/rules/blob/617b4272c7a07cfb20e3dc1c7aa8a0b1fcdc3c71/Emerging_Threats/blackmatter_ransomware.yml#L39

maximelb commented on Emerging_Threats/guloader.yml:
```diff
@@ -0,0 +1,155 @@
+  namespace: general
+  detect:
+    event: ONGOING_IDENTITY
+    op: ends with
```
Hashes are always a specific length, so we should use `is` for hashes.
```suggestion
    op: is
```

gio-pecora replied:
I agree,
I was using the as reference the previous template pushed/reviewed on the repo:
https://github.com/refractionPOINT/rules/blob/617b4272c7a07cfb20e3dc1c7aa8a0b1fcdc3c71/Emerging_Threats/blackmatter_ransomware.yml#L39

maximelb commented on Emerging_Threats/guloader.yml:
```diff
@@ -0,0 +1,155 @@
+      event: DNS_REQUEST
+      op: is
+      path: event/DOMAIN_NAME
+      value: nowautomation.com
```
Is it just this domain that's bad, or also possibly subdomains?
If subdomains should be included you could to:
```yaml
event: DNS_REQUEST
op: is
path: event/DOMAIN_NAME
value: nowautomation.com
sub domain: '-2:'
```

as documented here: https://doc.limacharlie.io/docs/documentation/ZG9jOjE5OTA0MjQ0-reference-operators#sub-domain

gio-pecora replied:
https://refractionpoint.slack.com/archives/C02DNHSF8JF/p1638549218167000?thread_ts=1638478408.157200&cid=C02DNHSF8JF

maximelb reviewed (changes requested):
Also generally, it might be nice to group the hashes relating to the same dropper together in one rule that uses OR, then the same for DNS relating to the same one.

Makes it nicer to have one logical rule that reports a general threat based on any number of indicators.

Commit:
Template optimisation

- Optimised subdomains detections
- Optimised HASH detection trough `is` operator and `CODE_IDENTITY` as part of the standard telemetry sent to the cloud.
